ENIGMAsystem/SHELL/Platforms/SDL/Gamepad.cpp

```diff
@@ -33,7 +40,7 @@ void pushGamepads() {
   for (Gamepad& gp : gamepads) gp.push();
 }
 
-void setGamepadButton(int gamepad, int btn, bool pressed) { gamepads[gamepad].state.buttonStatus[btn] = pressed; }
+void setGamepadButton(int gamepad, int btn, bool pressed) { gamepads[gamepad].state.buttonStatus[digitalButtons[btn]] = pressed; }
```

Comment:
This seems prone to going out of bounds. I think we should either bounds check or use a hash of the buttons so were not limited to the 15 buttons

Reply:
I don't know, we do this in a lot of places. Sure, using a real map would be the safest, but it's the most verbose to construct. I think that's why Josh favored just flat hash maps (aka arrays) in these places. We do this for graphics constants and other stuff everywhere. The only thing I've seen Josh do in the past is modulo the array size, but that's not really correct. Maybe it's time we make a ConstantArray to with AssetArray.

Reply:
Also, by the way, technically you can check the axes as if they are buttons, that's where threshold comes in. At a certain point the bumpers quick being analog and become digital buttons. I didn't feel like fixing that right now, just trying to fix basic user's error, as there's plenty more to do here including fix tkg's bug with unplugging.

Reply:
Issue I have is what you have now will crash. Ideally it should be impossible to crash your game in debug mode and we should be issuing hints to what did cause a crash in run mode. I would prefer a hash cause gm is not the arbiter of number of buttons on a controller but at the same time we could earn if a user was using a non standard value and insert it into the map without crash. However at the bare min we should check if the I'd is greater than 15 here if you just want the quick and dirty fix for now

Reply:
It won't crash _here_ fundies, this array is used by the function that the SDL event processing calls. The only way it would crash is if SDL gave us a bad value from its own enum... which would be... weird. That said, I can add it anyway, but it's not that big of a deal because I have to come back in here to fix the analog to digital axes anyway.

Reply:
Ok. I thought this was a user function. Although, I still wouldnt trust sdl's enum to remain in your bounds (15) forever. gamepads have more and more buttons each generation it seems.

Reply:
I thought about it and 451edc52a9891bcdfd3593d54a3ce2a0e6b00c76 you're right about that.